src/main/java/frc/robot/VisionFieldLayoutRecognizer.java

```diff
@@ -0,0 +1,88 @@
+package frc.robot;
+
+import java.util.ArrayList;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import frc.robot.AutonomousPath.VisionPathName;
+
+
+/*
+ * VisionFieldLayoutRecognizer
+ * 
+ * Version 1.0
+ *
+ * 02/27/2021
+ * 
+ * EnTech 
+ */
+
+ public class VisionFieldLayoutRecognizer
+    implements PixyFieldPoseResolver{
+
+     
+    
+    public VisionFieldLayoutRecognizer() {
+      
+    }
+
+     public AutonomousPath detectPose(ArrayList<FieldVisionInput> arraylistFieldVisionInput) {
+        
+      AutonomousPath objectAutonomousPath;
+      FieldVisionInput objectFieldVisionInput;
+      DistanceRange objectDistanceRange;
+
+      double dblFocalLength = 11.33;//pixels = 3mm
```

Comment:
when you find yourself writing comments, ask yourself if the code can tell the story.  even better, have the computer do the work. 

you should make these values constants ( all caps), and they need the data you have in comments as the name.

final const double PIXY_WIDTH_INCHES = 7.0;
final const double PIXY_CAMERA_FOCAL_LENGTH_INCHES = 620/3.0 ( or whatever calculation you did to get 11.33 out of 3mm )